packages/ethereum-storage/src/smart-contract-manager.ts

```diff
@@ -282,16 +289,63 @@ export default class SmartContractManager {
         // When set to true, we use it to ignore next confirmation event function call
         let ethereumMetadataCreated: boolean = false;
 
+        // Keep the transaction hash for future needs
+        let transactionHash: string = '';
+
         this.requestHashSubmitter.methods
           .submitHash(contentHash, feesParametersAsBytes)
           .send({
             from: account,
             gas: '100000',
             gasPrice: gasPriceToUse,
+            nonce,
             value: fee,
           })
-          .on('error', (transactionError: string) => {
-            reject(Error(`Ethereum transaction error:  ${transactionError}`));
+          .on('transactionHash', (hash: any) => {
+            // Store the transaction hash in case we need it in the future
+            transactionHash = hash;
+          })
+          .on('error', async (transactionError: string) => {
+            // If failed because of polling timeout, try to resubmit the transaction with more gas
+            if (
+              transactionError.toString().includes(TRANSACTION_POLLING_TIMEOUT) &&
+              transactionHash
+            ) {
+              // If we didn't set the nonce, find the current transaction nonce
+              if (!nonce) {
+                const tx = await this.eth.getTransaction(transactionHash);
+                nonce = tx.nonce;
+              }
+
+              // Get the new gas price for the transaction
+              const newGasPrice = new bigNumber(
+                await gasPriceDefiner.getGasPrice(StorageTypes.GasPriceType.FAST, this.networkName),
+              );
+
+              // If the new gas price is higher than the previous, resubmit the transaction
+              if (newGasPrice.gt(new bigNumber(gasPriceToUse))) {
+                // Retry transaction with the new gas price and propagate back the result
+                try {
+                  resolve(
+                    await this.addHashAndSizeToEthereum(
```

Comment:
It could be safer to use a retry method in Web3 but this works just as well!

Reply:
I wish I could use it, but web3.js does not implement this endpoint.
It's [specified in geth](https://github.com/ethereum/go-ethereum/blob/9371b2f70c70f887999d10750440fb09fe1e184f/internal/web3ext/web3ext.go#L485), but not web3.js